### ISSUE TITLE
Size of output model is half of original

### ISSUE THREAD

**Issue by bonadio** [MinishLab/model2vec#213]
Hello

I am trying to get a small model "Multi-language" to run on a Browser with Onnx.

I tried to run Model2vec on paraphrase-multilingual-MiniLM-L12-v2 (470MB) original 
```
from model2vec.distill import distill

m2v_model = distill(model_name="sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2", pca_dims=256)

# Save the model
m2v_model.save_pretrained("m2v_model")
``` 

The result model was 256MB. Is there anything I can do to make it smaller?  

The onnx quantized version of paraphrase-multilingual-MiniLM-L12-v2  model_quint8_avx2.onnx is 118MB

Thanks

**Comment by Pringled** [comment]
Hi @bonadio,

The easiest way to reduce the size of your model is reducing the output dimensionality further. For example:

```python
m2v_model = distill(model_name="sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2", pca_dims=128)
```

Will reduce your model size by another factor of 2. You can play around with this parameter to get a model size for your usecase. The onnx model you mentioned is quantized, hence the small size. We don't support quantization out of the box, but it's something you can apply on the weights yourself if you're interested in that. 

One note though: reducing the dimensionality will not make the model2vec model faster, it will only reduce the size. So for optimal performance, we recommend keeping the pca_dims around 256.

**Comment by stephantul** [comment]
Hi @bonadio,  

#217 adds a quantization option to model2vec. Once we merge this PR, models will be half the current size by default. We also added the option to quantize to int8, which should reduce the size to 25% of the current size, without loss of performance.

If you want, you can already check this out by cloning the relevant branch. I will notify you once we merge and/or release. Hope this helps!

**Comment by bonadio** [comment]
Hi @stephantul 
I will try that,  but the problem I found is that the generated model from model2vec has a large drop in accuracy, from 85% with the original onnx int8 quantized model  (paraphrase-multilingual-MiniLM-L12-v2)  to 65% with the model2vec with pca=256.  My eval is very simple just 2 phrases and 40 words calculating the cosine similarity.  The speed of model2vec is almost instantaneous.  

Here is the code:
```python 
from model2vec.distill import distill

m2v_model = distill(model_name="sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2", pca_dims=256)

# Save the model
m2v_model.save_pretrained("m2v_model_256")

from model2vec import StaticModel
import numpy as np
from sklearn.metrics.pairwise import cosine_similarity
from sentence_transformers import SentenceTransformer

model = StaticModel.from_pretrained("m2v_model_256")

def find_closest_documents(query_vector, document_embeddings, top_k=5):
    similarities = cosine_similarity([query_vector], document_embeddings)
    closest_indices = np.argsort(similarities[0])[::-1][:top_k]  
    return closest_indices.tolist()


sentences = ["Isso é um cachorro quente", "O submarino viaja no oceano azul"]

word_list = {
  "Isso é um cachorro quente": [
    "pão",
    "salsicha",
    "mostarda",
    "ketchup",
    "maionese",
    "lanche",
    "molho",
    "batata",
    "cebola",
    "picles",
    "sanduíche",
    "churrasco",
    "fast-food",
    "refeição",
    "queijo",
    "bacon",
    "molho barbecue",
    "comida",
    "salgado",
    "lanchonete"
  ],
  "O submarino viaja no oceano azul": [
    "mar",
    "água",
    "fundo",
    "navegação",
    "mergulho",
    "sonar",
    "tripulação",
    "náutico",
    "periscópio",
    "embarcação",
    "oceânico",
    "profundidade",
    "azul",
    "exploração",
    "torpedo",
    "marinha",
    "propulsão",
    "casco",
    "pressão",
    "silêncioso"
  ]
}
# Make embeddings
embeddings = model.encode(sentences)

correct_count = 0
total_count = 0

#evaluate model2vec

for sentence, words in word_list.items():
    for word in words:
        total_count += 1
        query_embedding = model.encode(word)
        closest_docs_indices = find_closest_documents(query_embedding, embeddings, top_k=1)
        print(f"word {word}")
        print(f"Closest document sentence: {sentences[closest_docs_indices[0]]}")
        if sentences[closest_docs_indices[0]] == sentence:
            print("correct")
            correct_count += 1
        else:
            print("incorrect")
        print(f"-"*20)

accuracy = (correct_count / total_count) * 100 if total_count > 0 else 0
print(f"Accuracy: {accuracy:.2f}%")

```

**Comment by stephantul** [comment]
Hey @bonadio,

You can bump your performance by adding specific words to your vocabulary during distillation:

```python

vocabulary = [
    "pão",
    "salsicha",
    "mostarda",
    "ketchup",
    "maionese",
    "lanche",
    "molho",
    "batata",
    "cebola",
    "picles",
    "sanduíche",
    "churrasco",
    "fast-food",
    "refeição",
    "queijo",
    "bacon",
    "molho barbecue",
    "comida",
    "salgado",
    "lanchonete",
    "mar",
    "água",
    "fundo",
    "navegação",
    "mergulho",
    "sonar",
    "tripulação",
    "náutico",
    "periscópio",
    "embarcação",
    "oceânico",
    "profundidade",
    "azul",
    "exploração",
    "torpedo",
    "marinha",
    "propulsão",
    "casco",
    "pressão",
    "silêncioso"
  ]

m2v_model = distill(model_name="sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2", pca_dims=256, vocabulary=vocabulary)
```

This will add those tokens to your vocabulary. Although I don't know how well this will generalize.
